yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtTree.kt

```diff
@@ -249,88 +422,55 @@ internal class CrdtTree(
-        } else {
-            allChildren.indexOf(leftSiblingNode) + 1
+        // 01. Find the parent and left sibling node of the given position.
+        val (parent, leftSibling) = pos.toTreeNodes(this)
+
+        // 02. Determine whether the position is left-most and the exact parent
+        // in the current tree.
+        val isLeftMost = parent == leftSibling
+        val realParent =
+            leftSibling.parent.takeIf { leftSibling.parent != null && !isLeftMost }
+                ?: parent
+
+        // 03. Split text node if the left node is a text node.
+        if (leftSibling.isText) {
+            leftSibling.split(this, pos.leftSiblingID.offset - leftSibling.id.offset)
         }
 
-        var updatedLeftSiblingNode = leftSiblingNode
-        for (i in index until parentNode.allChildren.size) {
-            val next = allChildren[i]
-            if (executedAt < next.id.createdAt) {
-                updatedLeftSiblingNode = next
-            } else {
-                break
-            }
+        // 04. Find the appropriate left node. If some nodes are inserted at the
+        // same position concurrently, then we need to find the appropriate left
+        // node. This is similar to RGA.
+        val allChildren = realParent.allChildren
+        val index = if (isLeftMost) 0 else allChildren.indexOf(leftSibling) + 1
+
+        var updatedLeftSiblingNode = leftSibling
+        for (node in allChildren.drop(index)) {
+            if (node.id.createdAt <= executedAt) break
+            updatedLeftSiblingNode = node
         }
-        return parentNode to updatedLeftSiblingNode
+        return realParent to updatedLeftSiblingNode
     }
 
-    private fun findFloorNode(id: CrdtTreeNodeID): CrdtTreeNode? {
+    fun findFloorNode(id: CrdtTreeNodeID): CrdtTreeNode? {
         val (key, value) = nodeMapByID.floorEntry(id) ?: return null
-        return if (key == null || key.createdAt != id.createdAt) null else value
-    }
-
-    private fun toTreeNodes(pos: CrdtTreePos): Pair<CrdtTreeNode, CrdtTreeNode>? {
-        val parentID = pos.parentID
-        val leftSiblingID = pos.leftSiblingID
-        val parentNode = findFloorNode(parentID) ?: return null
-        val leftSiblingNode = findFloorNode(leftSiblingID) ?: return null
-
-        val updatedLeftSiblingNode =
-            if (leftSiblingID.offset > 0 && leftSiblingID.offset == leftSiblingNode.id.offset &&
-                leftSiblingNode.insPrevID != null
-            ) {
-                findFloorNode(requireNotNull(leftSiblingNode.insPrevID)) ?: leftSiblingNode
-            } else {
-                leftSiblingNode
-            }
-        return parentNode to updatedLeftSiblingNode
+        return value.takeIf { key.createdAt == id.createdAt }
     }
 
     /**
      * Edits the given [range] with the given [contents].
      * This method uses indexes instead of a pair of [TreePos] for testing.
      */
-    fun editByIndex(
+    @VisibleForTesting
+    fun editT(
```

Comment:
what is the meaning of `editT`?

Reply:
I think `T` stands for `Test` since the function is only for testing.
the name `editT` was a bit confusing to me as well, and the function isn't that essential, so I just removed it in [b5998fd](https://github.com/yorkie-team/yorkie-android-sdk/pull/147/commits/b5998fd9a5c9022159383f5ba1f84947ee5c5f01).
it works the same though.

(cc. @hackerwins )